src/ImageSharp.Web/Resolvers/PhysicalFileSystemCacheResolver.cs

```diff
@@ -30,21 +30,23 @@ public PhysicalFileSystemCacheResolver(FileInfo metaFileInfo, FormatUtilities fo
         }
 
         /// <inheritdoc/>
-        public async Task<ImageCacheMetadata> GetMetaDataAsync()
-        {
-            using Stream stream = OpenFileStream(this.metaFileInfo.FullName);
-            this.metadata = await ImageCacheMetadata.ReadAsync(stream);
-            return this.metadata;
-        }
+        public async Task<ImageCacheMetadata> GetMetaDataAsync() => this.metadata ??= await this.ReadMetadataAsync();
```

Comment:
This is where potential the issue lies and the rest of the types should have been left untouched. We need to introduce a null check as you have but also try to reduce the overhead of that new check.